lib/utility/recipe_puppy_parser.rb

```diff
@@ -0,0 +1,74 @@
+module Utility
+  class RecipePuppyParser
+    BASE_URL = 'http://www.recipepuppy.com/api/'.freeze
+    ALL_RECIPES_URL = 'http://www.recipepuppy.com/api/?q=&p='.freeze
+    RECIPE_QUERY_BASE = 'http://www.recipepuppy.com/api/?q='.freeze
+    MAX_PAGES = 100.freeze
+
+    def query_for(recipe)
+      recipe.gsub!(/\s/, '+')
+      MAX_PAGES.times do |page|
+        searched_url = "#{RECIPE_QUERY_BASE}#{recipe}&p=#{page + 1}"
+        get_recipes_to_query(recipe, searched_url)
+      end
+    end
+
+    def query_all_recipes_for(*ingredients)
+      ingredient_query_syntax = ingredients.join('').gsub(/\s+/, "")
+      contact_recipe_api(ingredient_query_syntax)
+    end
+
+    def contact_recipe_api(ingredient_query)
+      recipes = []
+      MAX_PAGES.times do |page|
```

Comment:
Same kind of idea for this as in `query_for(recipe)`. Please frame it in terms of being used in a lightweight way for search and a heavyweight way in tasks.